src/main/scala/io/qbeast/spark/delta/IndexStatusBuilder.scala

```diff
@@ -28,10 +27,16 @@ private[delta] class IndexStatusBuilder(
    * Dataset of files belonging to the specific revision
    * @return the dataset of AddFile actions
    */
-  def revisionFiles: Dataset[AddFile] = {
+  def revisionFiles: Dataset[QbeastFile] = {
+    val spark = SparkSession.active
+    import spark.implicits._
+
     // this must be external to the lambda, to avoid SerializationErrors
     val revID = revision.revisionID.toString
-    qbeastSnapshot.snapshot.allFiles.filter(_.tags(TagUtils.revision) == revID)
+    qbeastSnapshot.snapshot.allFiles
+      .filter(_.tags(TagUtils.revision) == revID)
```

Comment:
filtering files by revID, should be part of the functionality of QbeastSnapshot. Or in other words, wouldn't be better if the field snapshot is private in QbeastSnapshot? So we don't rely on Delta's API too much.

Reply:
Yes, we can exclude the IndexStatusBuilder of using delta objects.